File: segmentManager/src/main/java/com/clumob/segment/manager/SegmentViewHolderFactory.java
package com.clumob.segment.manager;

import android.content.Context;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.ViewGroup;

public interface SegmentViewHolderFactory {
    SegmentViewHolder<?,?> create(Context context, LayoutInflater layoutInflater, @Nullable ViewGroup parentView);
}

File: segmentManager/src/main/java/com/clumob/segment/manager/SegmentViewHolderFactory.kt
package com.clumob.segment.manager

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup

interface SegmentViewHolderFactory {
    fun create(context: Context, layoutInflater: LayoutInflater, parentView: ViewGroup?): SegmentViewHolder<*, *>
}